src/index.js

```diff
@@ -30,7 +30,20 @@ function parse (args) {
 
     if (selector.classes !== '') {
       if (attrs.class) {
-        attrs.class = `${attrs.class} ${selector.classes}`
+        if (attrs.class instanceof Array) {
+          attrs.class = selector.classes.split(/[ ]+/g).concat(attrs.class)
+        } else if (typeof attrs.class === 'string') {
+          attrs.class = selector.classes + ' ' + attrs.class
+        } else if (typeof attrs.class === 'object') {
+          const newClassesObj = {}
+          selector.classes.split(/[ ]+/g).forEach(className => {
+            newClassesObj[className] = true
+          })
+          Object.keys(attrs.class).forEach(existing => {
+            newClassesObj[existing] = attrs.class[existing]
+          })
+          attrs.class = newClassesObj
```

Comment:
This is to preserve order. Classnames defined with tagname should come first (IMO).